include/swift/AST/Decl.h

```diff
@@ -2854,8 +2854,9 @@ class NominalTypeDecl : public GenericTypeDecl, public IterableDeclContext {
   ///
   /// \param ignoreNewExtensions Whether to avoid loading any new extension.
   /// Used by the module loader to break recursion.
-  ArrayRef<ValueDecl *> lookupDirect(DeclName name,
-                                     bool ignoreNewExtensions = false);
+  SmallVector<ValueDecl *, 4> lookupDirect(
```

Comment:
The result of this function is generally stored on the stack, so we're mostly talking saving some stack space (`TinyPtrVector` is one pointer vs. seven for the `SmallVector`). I'll make the change, though.